(Developers) Improve Dogshell docs (#24944)
DataDog/documentation: content/en/developers/guide/dogshell.md

---
title: Dogshell
description: "Use Datadog's API from Terminal or Shell"
aliases:
  - /developers/faq/dogshell-quickly-use-datadog-s-api-from-terminal-shell
  - /developers/guide/dogshell-quickly-use-datadog-s-api-from-terminal-shell
---

You can use the Datadog API on the command line using a wrapper called Dogshell.

## Install Dogshell

> Dogshell comes with the officially supported [datadogpy Python library][1], which is often used to send data to Datadog with [DogStatsD][2]. To install the library with PIP, run the following command:

Reviewer: ```suggestion
Dogshell comes with the officially supported [`datadogpy` Python library][1], which is often used to send data to Datadog with [`DogStatsD`][2]. To install the library with PIP, run the following command:
```

{{< code-block lang="shell" >}}
pip install datadog
{{< /code-block >}}

> Depending on your environment, you might have to add the library to your PATH. See the [datadogpy GitHub repo][3] for alternate installation instructions.

Reviewer: ```suggestion
Depending on your environment, you might have to add the library to your PATH. See the [`datadogpy` GitHub repo][3] for alternative installation instructions.
```

## Configure Dogshell

Dogshell uses a configuration file called `.dogrc` to store your API key, application key, and Datadog site.

To configure Dogshell:
1. Create a `.dogrc` file in your home directory:
   {{< code-block lang="shell" >}}
touch ~/.dogrc
{{< /code-block >}}

> 1. Add the following content to the file, replacing `DATADOG_API_KEY` and `YOUR_APPLICATION_KEY` with your API key and application key respectively:

Author: ```suggestion
1. Add the following content to the file, replacing `DD_API_KEY` and `DD_APP_KEY` with your API key and application key respectively:
```

   ```conf
   [Connection]
   apikey = DD_API_KEY

>    appkey = DD_APP_KEY

Reviewer: I think these should probably still have the angle brackets

Author: I just realized I don't like `DD_` here cos that's usually a reserved environment variable. I'm going to use `MY_` instead - we use that fairly often. I'm not a huge fan of angle brackets cos I find it sometimes confuses users - they enclose the values in them by mistake.

>    appkey = <YOUR_APPLICATION_KEY>

Reviewer: Don't we usually use DD_API_KEY and DD_APP_KEY?

Author: ```suggestion
   apikey = DD_API_KEY
   appkey = DD_APP_KEY
```

   api_host = https://{{< region-param key="dd_site">}}
   ```

>    <div class="alert alert-info">You can create multiple configuration files if you need to run commands against different environments. Use the <code>--config</code> flag to specify the path to an alternate configuration file.</div>

Reviewer: ```suggestion
   <div class="alert alert-info">You can create multiple configuration files if you need to run commands against different environments. Use the <code>--config</code> flag to specify the path to an alternative configuration file.</div>
```
